journal/src/main/java/io/zeebe/journal/file/MappedJournalSegmentWriter.java

```diff
@@ -88,9 +94,31 @@ public JournalRecord append(final long asqn, final DirectBuffer data) {
 
     // TODO: Should reject append if the asqn is not greater than the previous record
 
-    final int recordStartPosition = buffer.position();
-    lastEntry = write(buffer, recordIndex, asqn, data);
-    index.index(lastEntry, recordStartPosition);
+    final MutableDirectBuffer writeBuffer = new UnsafeBuffer();
```

Comment:
We will need a follow up issue to clean this method up - there's several improvements we could do (mostly always using a `MutableDirectBuffer`, safely managing the position/limit, etc.) but I guess it's out of scope.